File: src/main/java/com/drunkensoftball/api/exception/BadGatewayException.java
package com.drunkensoftball.api.exception;

import org.springframework.web.bind.annotation.ResponseStatus;

import static org.springframework.http.HttpStatus.BAD_GATEWAY;
import static org.springframework.http.HttpStatus.BAD_REQUEST;

@ResponseStatus(BAD_GATEWAY)
public class BadGatewayException extends RuntimeException {

    public BadGatewayException(final String message){
        super(message);
    }

    public BadGatewayException(final String message, final Throwable cause){
        super(message, cause);
    }
}

File: src/main/kotlin/com/drunkensoftball/api/exception/BadGatewayException.kt
package com.drunkensoftball.api.exception


import org.springframework.web.bind.annotation.ResponseStatus

import org.springframework.http.HttpStatus.BAD_GATEWAY

@ResponseStatus(BAD_GATEWAY)
class BadGatewayException : BaseDSException {

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}
}
